src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs

```diff
@@ -478,7 +480,7 @@ internal static PropertyInfo[] GetKeyPropertiesOnType(Type type, out bool hasPro
                 }
             }
 
-            if (newKeyKind == KeyKind.AttributedKey && keyProperties.Count != dataServiceKeyAttribute.KeyNames.Count)
+            if (newKeyKind == KeyKind.AttributedKey && keyProperties.Count != dataServiceKeyAttribute?.KeyNames.Count)

```

Comment:
@ElizabethOkerio Is there a specific reason for introducing the `?` operator on this line?

Reply:
Yes. So if someone used the property key attribute instead of the type attribute, then this statement will be executed  as the keykind is AttributedKey and the dataserviceKeyAttribute for a key property attribute is null. So I limited this check to only when a type attribute is used.